### PR TITLE
Object oriented programing

### LINKED ISSUE #38: Add dry-run
Collect links and print them Only Mode (without sending an req to wayback machine)
Like: `-d, --dry`

### LINKED ISSUE #39: Add wait randomly within searching
Need reduction of the burden toward target sites.

### LINKED ISSUE #43: Archive.is_end() is called only from Interact
Archive.is_end() is called only from Interact.chk_url_cond().
So I think this method should be moved to Interact.

Archive.is_end() は Interact.chk_url_cond() からしか使われていません。
そのため、Interact内に移動したほうがいいと思います。

### LINKED ISSUE #44: Archive.retry_ntime() seems wrong
in wbsv/Archive.py:74 in archive()

```python
retry_ntime(RETRY,
      try_archive(id_, len(uri_dic), uri),
      (id_, len(uri_dic))
```

in wbsv/Archive.py:34

```python
def retry_ntime(ret, func, c, uri):
```

You want to call `func` n ( or `ret` ) times? (If not, ignore this message.) (Sorry, I cannot understand `id_`. )

then write

```python
retry_ntime(RETRY,
    try_archive,
    (id_, len(uri_dic), uri) )
```

and

```python
def retry_ntime(ret, func, args, uri):
    for cnt in range(ret):
        func(*args)
```

### PR DESCRIPTION
**Pull request by yswallow**
to run, `$ python3 -m wbsv.main-refactor -d {{url}}`

Fully rewriting, and including fix of  #38 , #39 , #43 , #44

### CODE REVIEW

**Comment by eggplants** [comment]
LGTM, but I want to keep the original formatting when outputting.
Thank you for your refactoring with OOP soooooo much!!!

File: wbsv/ParseArgs.py [hunk 1]
```diff
@@ -46,6 +46,8 @@ def natural_num(n):
     parser.add_argument('-L', '--level', default=0,
                         type=natural_num, metavar='lv',
                         help='Set maximum recursion depth.')
+    parser.add_argument('-d', '--dry_run', action='store_true', default=False,
```
Comment (eggplants) [inline wbsv/ParseArgs.py:49]: `default=False` is useless. please remove it.

**Comment by eggplants** [comment]
And rename obsolete files  into `[filename].py.old` and renewed files into names derived from the original file name.
古いファイル名を`[filename].py.old`にして同機能を持った新しいファイル名をリネームしてください。

**Comment by eggplants** [comment]
Uh, once I've merged it, I'll rewrite it as I see fit. Are you OK?

File: wbsv/archiver.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        self.retry = opt["retry"]
+
+    def try_archive(self, url):
+        pass
+
+    def archive(self, url):
+        print("Saving: "+url)
+        result = self.retry_ntimes( self.try_archive, [url] )
+        self.add_result(result)
+        print( ("Saved: " if result else "Failed: ")+url)
```
Comment (eggplants) [inline wbsv/archiver.py:25]: `print( ("[+]Saved: " if result else "Failed: ")+url)`

File: wbsv/archiver.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+            self.fail_count += 1
+
+    def print_result(self):
+        #print("[+]FIN!: %s" % pageurl)
+        print("SAVE:", self.save_count, "FAIL:", self.fail_count)
+
+class Archiver(AbstractArchiver):
+    def try_archive(self, url):
+        try:
+            #print("[%s/%d]: Wait...    " % (id_, dic_size), end="\r")
```
Comment (eggplants) [inline wbsv/archiver.py:50]: Uncomment this.

File: wbsv/archiver.py [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+            archive_uri, exist_f = capture_or_cache(url)
+            #print("[%s/%d]:" % (id_, dic_size), end=" ")
```
Comment (eggplants) [inline wbsv/archiver.py:52]: Uncomment this.

**Review by eggplants — COMMENTED** [review]
Almost OK...

**Comment by yswallow** [comment]
OK.

And I guess merge this onto a branch (not master), and you modify it, and
merge to master.

一度別のブランチに入れて修正してからマスターに上げる方がいいと思います。

**Pull request merged**